### ISSUE TITLE
Hey man, I started training with ur dataset as it is written. But the ETA was about 7 days..... 

### ISSUE THREAD

**Issue by Simon-Beck12** [patrickisroman/recyclr-vision#1]
I started training as it is written. The ETA said it will take about 7 days to finish the training... 
Is there anyway I can shorten the time within 1 day or is the time just estimated time?

**Comment by patrickisroman** [comment]
The ETA provided by YOLACT is based on iteration count, not accuracy: https://github.com/dbolya/yolact/blob/master/train.py#L333

The iteration count is defined within the config.py file:
https://github.com/dbolya/yolact/blob/master/data/config.py#L668

You can stop the training once the network is saturated and learning falls off, even if it's not near `max_Iters` iterations. The model is autosaved every 1000 iterations and you can evaluate the model using each weights file.

**Comment by Simon-Beck12** [comment]
> The ETA provided by YOLACT is based on iteration count, not accuracy: https://github.com/dbolya/yolact/blob/master/train.py#L333
> 
> The iteration count is defined within the config.py file:
> https://github.com/dbolya/yolact/blob/master/data/config.py#L668
> 
> You can stop the training once the network is saturated and learning falls off, even if it's not near `max_Iters` iterations. The model is autosaved every 1000 iterations and you can evaluate the model using each weights file.

hey Thx again for this amazing explanation. I really don't want to bother u anymore , but I don't know how to visualize or at least know whether the network is saturated or learning falls off... Could u tell me some tips?

**Comment by patrickisroman** [comment]
No worries, I'm really glad you have interest in trash detection!

The train script will emit an output such as:

`[105] 1080 || B : 0.1596 | C: 0.091 | M : 0.235 | S: 0.069 | I: 0.029 | T: 0.583 || ETA: 56:52:19 || timer: 0.71963` 

The "T" value logged is the total combined loss function of the model at the current iteration. Ideally, as training continues, this number will drop, and you'd want this to get as low as possible, indicating that the loss is low, ie accuracy is higher.

YOLACT is copied into the directory recyclr/datasets/recyclr/yolact under the main package directory. If you change into this directory, you can run a command such as

`python eval.py --trained_model=weights/WEIGHT_FILE.pth --config=yolact_recyclr_config --video=path/to/video.mp4 --score_threshold=.8`

Where you replace WEIGHT_FILE.pth with the most recent weight file, and path/to/video.mp4 to the video you'd like to evaluate on.

I have a lot of changes coming to this repo to simplify the detector, expand the dataset, separate object detection and material detection, make training/evaluating easier, add really cool sorting, and allow for drone streaming functionality. Due to COVID and life, it's a bit delayed, but I expect to merge these changes in the next month or two if you keep an eye on it.

**Comment by Simon-Beck12** [comment]
Thx again. patrickisroman
So I have trained it until like 190,000 iterartion. But the evaluation result wasn't what I expected.
Although the input video is just random video from Youtube containing some recyclable stuffs, still result was pretty disastrous. It couldn't really detect recyclables or even if it did, it couldn't really label it correctly.  
I'm guessing the reason maybe because of so many labels comapred to datasets.

The 190,000 weight file had about **'0.9 T value'**. I'll be uploading image from
![output](https://user-images.githubusercontent.com/17828537/101888919-1ec7f280-3be2-11eb-8d8a-ad7d042dd2ff.jpg)

output video just in case.
I'd like to have some advice about this problem. Thx again!!!
